Add Expanded CMS Support

davidzech opened the pull request:
* Support BER Encoded CMS
* Support Attached CMS
* Support Signed Attributes CMS

davidzech commented on Package.swift:
```diff
@@ -76,7 +76,7 @@ let package = Package(
     package.dependencies += [
         .package(url: "https://github.com/apple/swift-crypto.git", "2.5.0" ..< "4.0.0"),
-        .package(url: "https://github.com/apple/swift-asn1.git", from: "1.0.0"),
+        .package(url: "https://github.com/apple/swift-asn1.git", revision: "c7e239b5c1492ffc3ebd7fbcc7a92548ce4e78f0"),
```
Waiting on a version bump from swift-asn1 to consume the new BER feature set.

davidzech replied:
Fixed in 8713872df9c0bb57d6d6c1cc83cdec5a10399aa2

davidzech commented:
Leaving this in draft until [apple/swift-asn1 ](http://github.com/apple/swift-asn1) is tagged with a new version with BER support.

dnadoba commented:
@davidzech we have just released [Swift ASN.1 1.1.0](https://github.com/apple/swift-asn1/releases/tag/1.1.0)

davidzech commented:
> @davidzech we have just released [Swift ASN.1 1.1.0](https://github.com/apple/swift-asn1/releases/tag/1.1.0)

Ack. Updated and removed Draft status @dnadoba

Lukasa commented:
@swift-server-bot add to allowlist

Lukasa commented:
Based on CI, you need to add this block to your new files that import `Foundation`:

```
#if canImport(Darwin)
import Foundation
#else
@preconcurrency import Foundation
#endif
```

davidzech commented:
> Based on CI, you need to add this block to your new files that import `Foundation`:
> 
> ```
> #if canImport(Darwin)
> import Foundation
> #else
> @preconcurrency import Foundation
> #endif
> ```

Ack. I likely need to update CMake again too.

Lukasa reviewed (approved):
API breakage is acceptable here.
